crates/codegen/src/python.rs

```diff
@@ -42,7 +42,7 @@ impl CodeGen {
                 lvalue,
                 node.rvalue
             ));
-            for succ in self.graph.succ(idx).collect::<Vec<_>>() {
+            for succ in self.graph.succs(idx).collect::<Vec<_>>() {
                 self.work(succ);
             }
         } else if let Some(node) = FuncNode::concrete_mut(node) {
```

Comment:
Why are these `if let Some(node)` not a `match`?

Reply:
How would I do it with match?

Reply:
I'm unsure what `concrete_mut` does for each Node class, but I'm assuming the purpose of these if statements are to determine what type `node` (line 33) is. 

I feel a match construct would be a lot nicer as it reduces the need to write `if let Some(node)` so many times:

```rust
match node.type {
    FuncNode => {
        // Logic
    }
    _ => {
        panic!("invalid node type")
    }
}
```

Reply:
Assuming you're wanting `node.type` to be an enum, there are problems with how new node variants can be added (as enums have a fixed set of variants). 
Currently, any downstream user can create a new node struct, implement `NodeLike`, and add it to the graph. This doesn't work for enums, as that would require every upstream user to handle the case of the new variant.